xdsl/dialects/csl.py

```diff
@@ -883,6 +925,527 @@ def verify_(self) -> None:
+            (DsdType, i16_value),
+            (DsdType, u16_value),
+        ]
+
+
+@irdl_op_definition
+class Add16Op(_SymmetricBinary16BitOp):
+    """
+    Implements @add16 and @addc16
+    """
+
+    name = "csl.add16"
+    with_carry = opt_prop_def(BoolAttr)
+
+
+@irdl_op_definition
+class And16Op(_SymmetricBinary16BitOp):
+    """
+    Implements @and16
+    """
+
+    name = "csl.and16"
+
+
+@irdl_op_definition
+class ClzOp(_Unary16BitOp):
+    """
+    Implements @clz
+    """
+
+    name = "csl.clz"
+
+
+@irdl_op_definition
+class CtzOp(_Unary16BitOp):
+    """
+    Implements @ctz
+    """
+
+    name = "csl.ctz"
+
+
+@irdl_op_definition
+class FabsOp(_MultiprecisionDsdOp):
+    """
+    Implements @fabsh and @fabss
+    """
+
+    name = "csl.fabs"
+
+    def get_signatures(self) -> FunctionSignatures:
+        match self.precision.data:
+            case BuiltinOpPrecisionKind.half:
+                return [
+                    (DsdType, DsdType),
+                    (DsdType, Float16Type),
+                ]
+            case BuiltinOpPrecisionKind.single:
+                return [
+                    (DsdType, DsdType),
+                    (DsdType, Float32Type),
+                ]
+            case _:
+                return []
+
+
+@irdl_op_definition
+class FaddOp(_MultiprecisionDsdOp):
+    """
+    Implements @faddh, @faddhs, and @fadds
+    """
+
+    name = "csl.fadd"
+
+    def get_signatures(self) -> FunctionSignatures:
+        match self.precision.data:
+            case BuiltinOpPrecisionKind.half:
+                return [
+                    (DsdType, DsdType, DsdType),
+                    (DsdType, Float16Type, DsdType),
+                    (DsdType, DsdType, Float16Type),
+                    (f16_pointer, Float16Type, DsdType),
+                ]
+            case BuiltinOpPrecisionKind.mixed:
+                return [
+                    (DsdType, DsdType, DsdType),
+                    (DsdType, Float16Type, DsdType),
+                    (DsdType, DsdType, Float16Type),
+                    (f32_pointer, Float32Type, DsdType),
+                ]
+            case BuiltinOpPrecisionKind.single:
+                return [
+                    (DsdType, DsdType, DsdType),
+                    (DsdType, Float32Type, DsdType),
+                    (DsdType, DsdType, Float32Type),
+                    (f32_pointer, Float32Type, DsdType),
+                ]
+
+
+@irdl_op_definition
+class Fh2sOp(_BuiltinDsdOp):
+    """
+    Implements @fh2s
+    """
+
+    name = "csl.fh2s"
+
+    def get_signatures(self) -> FunctionSignatures:
+        return [
+            (DsdType, DsdType),
+            (DsdType, Float16Type),
+        ]
+
+
+@irdl_op_definition
+class Fh2xp16Op(_BuiltinDsdOp):
+    """
+    Implements @fh2xp16
+    """
+
+    name = "csl.fh2xp16"
+
+    def get_signatures(self) -> FunctionSignatures:
+        return [(DsdType, DsdType), (DsdType, Float16Type), (i16_pointer, Float16Type)]
```

Comment:
Can you add a comma at the end of this array, so that `black` splits it across the lines like it does with others?